src/Core/Test/Integration/Traits/SnapshotTesting.php

```diff
@@ -3,39 +3,79 @@
 namespace Shopware\Core\Test\Integration\Traits;
 
 /**
- * This trait allows to run an assertion on a stored snapshot.
- * During normal test operation the $actual parameter is ignored,
- * and the data stored in the snapshot file is used.
- * But when the env variable 'UPDATE_SNAPSHOTS' is set to 1,
- * the $actual parameter is written to the snapshot file.
+ * Trait to snapshot test JSON arrays or Twig-rendered HTML strings.
+ *
+ * On first run (UPDATE_SNAPSHOTS=1), writes the snapshot file (.json or .html).
+ * On subsequent runs, asserts current output matches the stored snapshot.
  *
  * @internal
  */
 trait SnapshotTesting
 {
     /**
-     * @param array<string, mixed>|list<mixed> $actual
+     * @param array<mixed>|string $actual Array data to encode as JSON, or HTML string.
+     *
+     * @throws \JsonException
      */
-    private function assertSnapshot(string $expectedSnapshotName, array $actual, string $message = ''): void
+    private function assertSnapshot(string $name, array|string $actual, string $message = ''): void
     {
-        $class = new \ReflectionClass(static::class);
-        $baseDir = \dirname((string) $class->getFileName());
-        $snapshot = $baseDir . '/_snapshots/' . $expectedSnapshotName . '.json';
-        $envVar = $_SERVER['UPDATE_SNAPSHOTS'] ?? '';
-        $runUpdate = $envVar !== 'false'
-            && $envVar !== '-1'
-            && $envVar !== '';
-
-        if ($runUpdate) {
-            file_put_contents(
-                $snapshot,
-                json_encode($actual, \JSON_PRETTY_PRINT | \JSON_THROW_ON_ERROR | \JSON_UNESCAPED_SLASHES)
+        $snapshotDir = $this->getSnapshotDirectory();
+
+        $isHtml = \is_string($actual);
+        $extension = $isHtml ? 'html' : 'json';
+        $filePath = "$snapshotDir/$name.$extension";
+
+        $update = ($_SERVER['UPDATE_SNAPSHOTS'] ?? '') === '1';
```

Comment:
In my opinion the old way was better (just checking for the disabled values), because it allowed you to just set the env var to any value (e.g. `true`, `yes`, `1`, ...) or to turn it off not specify it at all or specifically setting it to `false`, `-1` (which should be much rarer in practice).

By just checking if it is exactly `1` you can easily set the env var wrong and wonder why it doesn't work (e.g. setting it to `true`).